src/MizarWidgets/include/MizarWidgets/SamplingWithFrameTrackInputWidget.h

```diff
@@ -5,30 +5,92 @@
 #ifndef MIZAR_WIDGETS_SAMPLING_WITH_FRAME_TRACK_INPUT_WIDGET_H_
 #define MIZAR_WIDGETS_SAMPLING_WITH_FRAME_TRACK_INPUT_WIDGET_H_
 
+#include <absl/container/flat_hash_map.h>
+#include <absl/container/flat_hash_set.h>
+#include <stdint.h>
+
+#include <QLabel>
+#include <QListWidget>
+#include <QObject>
 #include <QWidget>
 #include <memory>
 #include <string_view>
+#include <vector>
 
 #include "MizarData/MizarPairedData.h"
+#include "MizarData/SamplingWithFrameTrackComparisonReport.h"
 
 namespace Ui {
 class SamplingWithFrameTrackInputWidget;
 }
 
 namespace orbit_mizar_widgets {
 
-class SamplingWithFrameTrackInputWidget : public QWidget {
+class SamplingWithFrameTrackInputWidgetBase : public QWidget {
+  Q_OBJECT
  public:
-  explicit SamplingWithFrameTrackInputWidget(QWidget* parent);
-  ~SamplingWithFrameTrackInputWidget() override;
+  ~SamplingWithFrameTrackInputWidgetBase() override;
+
+  [[nodiscard]] orbit_mizar_data::HalfOfSamplingWithFrameTrackReportConfig MakeConfig() const;
 
-  template <typename PairedData = orbit_mizar_data::MizarPairedData>
-  void Init(const PairedData* data, const QString& name);
+ public slots:
+  void OnThreadSelectionChanged();
 
  private:
   std::unique_ptr<Ui::SamplingWithFrameTrackInputWidget> ui_;
+
+ protected:
+  explicit SamplingWithFrameTrackInputWidgetBase(QWidget* parent);
+
+  [[nodiscard]] QLabel* GetTitle() const;
+  [[nodiscard]] QListWidget* GetThreadList() const;
+
+  absl::node_hash_map<std::unique_ptr<QListWidgetItem>, uint32_t> tid_list_widget_items_to_tids_;
```

Comment:
I think I would go a different way than this map here. The reason is that the lookup that you use in the cpp is suspicious (`.at(QListWidgetItem*)` seems to work, but the types are not the same).

I would subclass `QListWidgetItem` and give it a field for thread id. Then you dont need this map here, and the other problem with the ownership also goes away. You can take a look at  `OverrideMappingItem` in SymbolLocationsDialog.cpp

Reply:
`.at(QListWidgetItem*)` should be fine, as `operator==` and hash is guaranteed to be the same for unique_ptr and the raw ptr. 

>I would subclass QListWidgetItem and give it a field for thread id. 

But I would have to downcast the `QListWidgetItem*` to `QTidListWidgetItem*` which IMO is more suspicious.

Reply:
I am going to close this comment and continue in the other one